### PR TITLE
Refactor/drop global state

### LINKED ISSUE #27: Get rid of global state in the implementation
As for now it should be enough to extract the following classes:
- `GitContext`, as a lazily evaluated cache of config, logs, reflogs, merge-bases etc.
- `MacheteContext`, for storing the branch dependency tree structure
- `CommandLineContext`, for options/command/args
- possibly `SystemContext` for checking/caching things like `isatty`, hook executability etc.

### PR DESCRIPTION
**Pull request by mkondratek**
- [x] CommandLineContext (done should be split b/w the rest)
- [ ] GitContext
- [x] MacheteContext
- [ ] SystemContext

### CODE REVIEW

File: git_machete/cmd.py [hunk 1]
```diff
@@ -115,8 +115,37 @@ def right_arrow() -> str:
     return "->" if ascii_only else u"➔"
 
 
-def ask_if(msg: str, opt_yes_msg: Optional[str], apply_fmt: bool = True) -> str:
-    if opt_yes and opt_yes_msg:
+class CommandLineContext:
```
Comment (PawelLipski) [inline git_machete/cmd.py:120]: I think this changeset can be merged as a self-contained PR, and other planned changes can go on separate PRs

File: git_machete/cmd.py [hunk 2]
```diff
@@ -2307,59 +2339,60 @@ def print_new_line(new_status: bool) -> None:
...
+    if cli_ctxt.opt_return_to == "here" or (cli_ctxt.opt_return_to == "nearest-remaining" and nearest_remaining_branch == initial_branch):
         print(f"Returned to the initial branch {bold(initial_branch)}")
-    elif opt_return_to == "nearest-remaining" and nearest_remaining_branch != initial_branch:
+    elif cli_ctxt.opt_return_to == "nearest-remaining" and nearest_remaining_branch != initial_branch:
         print(
             f"The initial branch {bold(initial_branch)} has been slid out. Returned to nearest remaining managed branch {bold(nearest_remaining_branch)}")
 
 
-def status(warn_on_yellow_edges: bool) -> None:
+def status(cli_ctxt: CommandLineContext, warn_on_yellow_edges: bool) -> None:
```
Comment (PawelLipski) [inline git_machete/cmd.py:2400]: Probably most of these methods will require all other contexts to be passed as arguments as well...

Comment (PawelLipski) [reply git_machete/cmd.py:2400]: Maybe wrapping then into single unified `Context` with `cli`/`machete`/`git` etc. fields won't be a bad idea

Comment (mkondratek) [reply git_machete/cmd.py:2400]: sounds good but let's leave the more specific name for now since the other context will be added on the separate PR(s)

**Review by PawelLipski — APPROVED** [review]
